libs/testserver/fake_workspace.go

```diff
@@ -81,6 +81,19 @@ func (s *FakeWorkspace) WorkspaceFilesImportFile(path string, body []byte) {
 		path = "/" + path
 	}
 	s.files[path] = body
+
+	// Add all directories in the path to the directories map
+	parts := strings.Split(path, "/")
```

Comment:
FWIW, you can also use [`path.Dir`](https://pkg.go.dev/path#Dir) to get the path leading up to the file. Then, you don't need to omit the last entry in the loop below. It also returns a cleaned result so no empty chunks.

Reply:
Addressed in  #2621